GitUI/CommitInfo/CommitInfo.cs

```diff
@@ -801,20 +801,84 @@ protected override void DisposeCustomResources()
+
+                foreach (string branch in branches)
+                {
+                    _orderByBranch[branch] = GetBranchOrder(branch);
+                }
+
+                return;
+
+                // Get the order for each branch.
+                // Add max possible order value to next "level" to sort properly with the order for each regex.
+                int GetBranchOrder(string branch)
+                {
+                    return branch == _currentBranch ? 0
+                        : 1
+                        + (
+                            TryGetOrder(branch, localBranchRegexes, out int localBranchOrder)
+
+                            // Matching local branch
+                            ? localBranchOrder
+                            : localBranchRegexes.Length
+                            + (
+                                TryGetOrder(branch, remoteBranchRegexes, out int remoteBranchOrder)
+                                ? (
+                                    TryGetOrder(branch, remoteRegexes, out int remoteOrder)
+
+                                    // prioritized branch, order per branch prio then remote
+                                    ? ((remoteBranchOrder * remoteRegexes.Length) + remoteOrder)
+
+                                    // prioritized branch but not remote, order after all prioritized remotes
+                                    : ((remoteBranchRegexes.Length * remoteRegexes.Length) + remoteBranchOrder))
+                                : (remoteBranchRegexes.Length * (remoteRegexes.Length + 1))
+                                + (
+                                    IsLocalBranch()
+
+                                    // Not prioritized local branch
+                                    ? 0
+                                    : TryGetOrder(branch, remoteRegexes, out remoteOrder)
+
+                                    // Not prioritized branch on prioritized remote (note that remoteOrder was not evaluated previously)
+                                    ? 1 + remoteOrder
+
+                                    // Non match, return int.Max
+                                    : int.MaxValue - 1 - localBranchRegexes.Length - (remoteBranchRegexes.Length * (1 + remoteRegexes.Length)))));
+
+                    bool IsLocalBranch() => !branch.StartsWith(_remoteBranchPrefix);
+
+                    bool TryGetOrder(string branch, string[] regexes, out int order)
+                    {
+                        int currentOrder = 0;
+                        foreach (string regex in regexes)
+                        {
+                            if (Regex.IsMatch(branch, regex, RegexOptions.ExplicitCapture))
+                            {
+                                order = currentOrder;
+                                return true;
+                            }
+
+                            currentOrder++;
+                        }
+
+                        order = int.MaxValue;
```

Comment:
```suggestion
                        order = currentOrder;
```
could save the accesses to `.Length` in the distinction of cases above. But since this cannot be used in all cases, it reads clearer with `.Length`.

Reply:
using that in some situations